Makefile

```diff
@@ -84,3 +91,24 @@ push-release:
 .PHONY: push
 push:
 	docker push $(IMAGE):latest
+
+.PHONY: generate-kustomize
+generate-kustomize: bin/helm
```

Comment:
This seems a bit over complicated. Is it possible to simply this step? I imagine it will be hard to update this list of command, whenever there is some changes or refactoring to the helm templates. I think it is okay to restructure the layer out of kustomize file structure to make the generation earlier and more maintainable. Maybe combine some of the kustomize files into one for each layer? As they are all auto generated.

Reply:
Yeah, I don't like it much either.  What I ran into is that when running helm template for files with more than one resource those resources weren't rendered in a consistent order.  I could run the target twice in a row and get the same resources but in a different order which makes it difficult to manage in source control.  I would be open if anyone has a way to produce reliably reproducible output and keep the alpha features separate in the folder structure.

Reply:
Given the limitation with helm. I think I can live with it for now, and we can follow up with this  as a separate issue.

Reply:
I just noticed this. It should have been fixed in Helm 3.1.0: https://github.com/helm/helm/pull/6842. Weird you were seeing this with Helm 3.1.2 here.

Reply:
Yeah, I saw that helm issue when I was working on this but still saw issues with resources moving around in the processed output on multiple runs without changing anything.